test/unit/useClientRequest.test.js

```diff
@@ -1,5 +1,97 @@
-import { useClientRequest } from '../../src';
+import React from 'react';
+import { act, testHook } from 'react-testing-library';
+import { useClientRequest, ClientContext } from '../../src';
+import { pathToFileURL } from 'url';
+
+let mockClient;
+
+const Wrapper = props => (
+  <ClientContext.Provider value={mockClient}>
+    {props.children}
+  </ClientContext.Provider>
+);
+
+const TEST_QUERY = `query Test($limit: Int) {
+  tests(limit: $limit) {
+    id
+  }
+}`;
 
 describe('useClientRequest', () => {
-  it('runs a test', () => {});
+  beforeEach(() => {
+    mockClient = {
+      getCacheKey: jest.fn().mockReturnValue('cacheKey'),
+      cache: {
+        get: jest.fn(),
+        set: jest.fn()
+      },
+      request: jest.fn().mockResolvedValue('bla')
+    };
+  });
+
+  it('returns a fetch function & state', () => {
+    let fetchData, state;
+    testHook(() => ([fetchData, state] = useClientRequest(TEST_QUERY)), {
+      wrapper: Wrapper
+    });
+    expect(fetchData).toEqual(expect.any(Function));
+    expect(state).toEqual({ cacheHit: false, loading: true });
+  });
+
+  describe('initial state', () => {
+    it('includes the cached response if present', () => {
+      mockClient.cache.get.mockReturnValueOnce({ some: 'cached data' });
+      let _, state;
+      testHook(() => ([_, state] = useClientRequest(TEST_QUERY)), {
```

Comment:
You don't need the `_` assignment here. You can do `[, state] = useClient...`.

Same thing on further down tests.